mars/services/task/execution/ray/executor.py

```diff
@@ -368,6 +371,74 @@ class _RayMonitorContext:
     )
 
 
+@dataclass
+class _RaySubtaskRuntime:
+    start_time: float = 0.0
+
+
+class _RaySlowSubtaskChecker:
+    @dataclass
+    class _CheckInfo:
+        count: int
+        upper: float
+
+    def __init__(
+        self,
+        total_subtask_count: int,
+        submitted_subtasks: OrderedSet,
+        completed_subtasks: OrderedSet,
+    ):
+        self._total_subtask_count = total_subtask_count
+        self._submitted_subtasks = submitted_subtasks
+        self._completed_subtasks = completed_subtasks
+        self._logic_key_to_subtask_costs = collections.defaultdict(list)
+        self._logic_key_to_check_info = dict()
+
+    def update(self):
+        i = 0
+        j = 0
+        while i < self._total_subtask_count or j < self._total_subtask_count:
+            curr_time = time.time()
+            while i < len(self._submitted_subtasks):
+                subtask = self._submitted_subtasks[i]
+                subtask.runtime.start_time = curr_time
+                i += 1
+            while j < len(self._completed_subtasks):
+                subtask = self._completed_subtasks[j]
+                self._logic_key_to_subtask_costs[subtask.logic_key].append(
+                    curr_time - subtask.runtime.start_time
+                )
+                j += 1
+            yield
+
+    def is_slow(self, subtask: Subtask):
+        logic_key = subtask.logic_key
+        if logic_key not in self._logic_key_to_subtask_costs:
+            # The subtask logic key has no costs.
+            return False
+        logic_parallelism = subtask.logic_parallelism
+        if not logic_parallelism:
+            # Invalid parallelism.
+            return False
+        subtask_costs = self._logic_key_to_subtask_costs[logic_key]
+        complete_count = len(subtask_costs)
+        if complete_count / logic_parallelism < 0.75:
```

Comment:
Could we turn the `0.75` to a configurable parameter?

Reply:
It could be a config, but if covers most cases, fewer config is better. If it is a config, users need to know the DAG to change the config.